usort/sorting.py

```diff
@@ -199,7 +199,24 @@ def sortable_blocks(
     return ret
 
 
-def is_sortable_import(stmt: cst.CSTNode) -> bool:
+def is_side_effect_import(base: str, names: List[str], config: Config) -> bool:
```

Comment:
I'd favor an implementation that's a little more straightforward in how it handles inputs and outputs, and doesn't need to locally construct the candidates (we won't need that anywhere else, AFAICT).  What do you think about having something like:

```py
class Config:
    def __post_init__(self):
        self.side_effect_re = "|".join( x + r"\b" for x in self.side_effect_modules )

def is_side_effect_import(base, names, config):
    return any(
        config.side_effect_re.match(
            some_testable_join_function(base, name)
        ) for name in names
    )
```

Reply:
also consider moving `is_side_effect_import` to Config, `def category` already is.